WIP: Add test for PostgreSQL restart

max-vostrikov opened the pull request:
a new test to check if clickhouse will reconnect if postgres server is killed.


<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ] <!---ci_set_required--> Allow: All Required Checks
- [ ] <!---ci_include_stateless--> Allow: Stateless tests
- [ ] <!---ci_include_stateful--> Allow: Stateful tests
- [x] <!---ci_include_integration--> Allow: Integration Tests
- [ ] <!---ci_include_performance--> Allow: Performance tests
- [ ] <!---ci_set_builds--> Allow: All Builds
- [ ] <!---batch_0_1--> Allow: batch 1, 2 for multi-batch jobs
- [ ] <!---batch_2_3--> Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ] <!---ci_exclude_style--> Exclude: Style check
- [ ] <!---ci_exclude_fast--> Exclude: Fast test
- [ ] <!---ci_exclude_asan--> Exclude: All with ASAN
- [ ] <!---ci_exclude_tsan|msan|ubsan|coverage--> Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ] <!---ci_exclude_aarch64|release|debug--> Exclude: All with aarch64, release, debug
---
- [ ] <!---do_not_test--> Do not test
- [ ] <!---woolen_wolfdog--> Woolen Wolfdog
- [ ] <!---upload_all--> Upload binaries for special builds
- [ ] <!---no_merge_commit--> Disable merge-commit
- [ ] <!---no_ci_cache--> Disable CI cache

al13n321 committed:
Disallow creating refreshable MV on Linux < 3.15

al13n321 committed:
Fix error message

max-vostrikov committed:
test_postgres_database_engine_restart test
a new test to check if clickhouse will reconnect if postgres server is killed

robot-clickhouse committed:
Automatic style fix

robot-ch-test-poll1 commented:
<!-- automatic status comment for PR #68033 from ClickHouse/ClickHouse:test_postgres_kill -->
*This is an automated comment for commit a58dc494ea82cd3fd55ea0a5214efc9c59f230e0 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68033/a58dc494ea82cd3fd55ea0a5214efc9c59f230e0/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68033/a58dc494ea82cd3fd55ea0a5214efc9c59f230e0/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

rschu1ze commented on tests/integration/helpers/cluster.py:
```diff
@@ -2056,6 +2056,17 @@ def restart_instance_with_ip_change(self, node, new_ip):
 
+    def restart_container(self, container_name):
+        p = subprocess.Popen(
+            ("docker", "restart", "-s", "9", container_name),
```
To avoid that future readers will need to look up what `-s` means (yeah, it _probably_ means "signal" but it is not 100% clear), please write `--signal` out.

rschu1ze replied:
Nit: by convention in this file, parameters ("docker", "restart" etc.) are written:

```py
(
    "docker",
    "restart",
    [...]
)
```

rschu1ze commented on tests/integration/helpers/cluster.py:
```diff
@@ -2056,6 +2056,17 @@ def restart_instance_with_ip_change(self, node, new_ip):
     def restart_service(self, service_name):
         run_and_check(self.base_cmd + ["restart", service_name])
 
+    def restart_container(self, container_name):
```
Perhaps existing function "run_and_check" will do the job already? (as helpers/cluster.py is generic framework code, I'd be reluctant with adding too much stuff that is called only from a single place)

rschu1ze commented on tests/integration/test_postgresql_database_engine/test.py:
```diff
@@ -465,6 +468,48 @@ def test_inaccessible_postgresql_database_engine_filterable_on_system_tables(
+            )
+        except Exception as e:
+            logging.debug(f"connection failed: {e}")
+            time.sleep(5)
```
So we will sleep 5 + 5 = 10 sec (l. 502 and l. 495) between iterations (l. 494), right?

max-vostrikov committed:
added ssl connection to postgres
the test is now connects to postgres using ssl to reproduce a bug https://github.com/ClickHouse/ClickHouse/issues/67069
moved the test to a spearate suite because ot was failing on parallel run

robot-clickhouse committed:
Automatic style fix

max-vostrikov commented:
the test is currently failing because of the underlying issue - https://github.com/ClickHouse/ClickHouse/issues/67069

1on committed:
Ability to limit columns for tables in MaterializedPostgreSQL

Commit:
Use another error code for external data in buffer

Use CANNOT_DECOMPRESS error code for the compressed external data in
CompressedReadBufferBase for checksum validation and decompression.

vitlibar committed:
Fix restoring access entities dependant on existing ones.

vdimir committed:
Randomize integration tests settings

vdimir committed:
comment

vitlibar committed:
Add test.

rschu1ze committed:
Remove superfluous --multiquery/-n

lgbo-ustc committed:
have_compressed is lost in reuseJoinedData

davenger committed:
Test with sparse serialization, vertical merge and skip indices

davenger committed:
Add materialization when building indices in vertical merge

tavplubix committed:
Merge pull request #69422 from ClickHouse/add_a_setting

Add a setting to disallow DETACH PERMANENTLY in Replicated database

kssenii committed:
Merge pull request #69454 from ClickHouse/revert-s3queue-problem

Quick fix for s3queue problem

yariks5s committed:
Update 03236_squashing_high_memory.sql

mstetsyuk committed:
do not fix metadata_version if replica is read_only

taiyang-li committed:
Merge branch 'ClickHouse:master' into orc_dict_encode

canhld94 committed:
remove unwanted include

Signed-off-by: Duc Canh Le <duccanh.le@ahrefs.com>

Felixoid committed:
Another attempt to address EAGAIN "Resource unavailable"

rschu1ze committed:
Merge pull request #69451 from rschu1ze/sqid-crash

Fix crash in `sqidDecode`

yariks5s committed:
Merge pull request #69455 from yariks5s/group_concat_return_string

`groupConcat` consistency with `arrayStringConcat`

yariks5s committed:
Update 03236_squashing_high_memory.sql

ucasfl committed:
Speedup insert data with vector similarity index by add data to index parallel

Michicosun committed:
add early break for test keeper impl

ucasfl committed:
remove blank line

yariks5s committed:
remove sanitizer builds

nickitat committed:
Merge pull request #69449 from ClickHouse/fix_test2

Fix test_disks_app_func

Felixoid committed:
Escape the `\n` in statements

kssenii committed:
Update aspell-ignore

Felixoid committed:
Merge pull request #69483 from ClickHouse/logs-export-improvement

Another attempt to address EAGAIN "Resource unavailable"

kssenii committed:
Merge pull request #69432 from ClickHouse/add-documnetation-for-azure-queue

Add azure-queue.md

devcrafter committed:
Merge branch 'master' into pr-local-plan

aiven-sal committed:
Allow SHOW COLUMNS to work with tables that contain dots in the name

aiven-sal committed:
Allow SHOW INDEXES to work with tables that contain dots in the name

rschu1ze committed:
CMake: Add comment about ICU data files

antonio2368 committed:
Merge pull request #66648 from ClickHouse/read-cgroup-memory-usage-async-metrics

Simplify some memory tracking parts

kssenii committed:
Move creation and validation of table metadata before creating queue metadata & save to factory

aiven-sal committed:
Add tests for SHOW COLUMNS from tables with dots

aiven-sal committed:
Add tests for SHOW INDEX from tables with dots

davenger committed:
Backport changes from private to minimize merge conflicts

rschu1ze committed:
Merge pull request #69511 from rschu1ze/icu-comment

CMake: Add comment about ICU data files

joelynch committed:
Properly clean up

davenger committed:
Merge pull request #69167 from ClickHouse/query_plan_for_merge

Use QueryPlan for merge

Commit:
Fix check in test_broken_projections

davenger committed:
Merge branch 'master' of github.com:ClickHouse/ClickHouse into tmp_files_lookup

kssenii committed:
Update test

robot-clickhouse committed:
Automatic style fix

antonio2368 committed:
Merge pull request #69469 from yariks5s/squashing_mem_usage

Fix high memory consumption on Squashing

marco-vb committed:
Add settings min_free_disk_bytes_to_throw_insert and min_free_disk_ratio_to_throw_insert and update documentation.

qoega committed:
merge remote-tracking branch 'origin/master' into restore-replace-to-null-settings-fix

kssenii committed:
Merge branch 'master' into s3queue-refactor

jkartseva committed:
Merge pull request #60341 from KevinyhZou/improve_join_insert_from

Improve left/inner join performance by rerange right table by keys

nickitat committed:
use final task size in segment size calculation

Felixoid committed:
Get rid of broken `get_commits().reversed`

evillique committed:
Fix attach of ReplicatedMergeTree tables in Replicated databases

devcrafter committed:
Merge pull request #64448 from ClickHouse/pr-local-plan

Local plan for parallel replicas

evillique committed:
Add test

robot-clickhouse committed:
Automatic style fix

yakov-olkhovskiy committed:
make it compatible with multiple runs

ucasfl committed:
Make vector similarity index creation thread pool globally

yakov-olkhovskiy committed:
fix

Felixoid committed:
Merge pull request #69538 from ClickHouse/fix-latest-sync-commit

Get rid of broken `get_commits().reversed`

yakov-olkhovskiy committed:
bump

ucasfl committed:
Merge branch 'master' of github.com:ClickHouse/ClickHouse into vector-index-insert

davenger committed:
Merge branch 'master' of github.com:ClickHouse/ClickHouse into tmp_files_lookup

davenger committed:
Merge branch 'master' of github.com:ClickHouse/ClickHouse into backport_mergetask

Michicosun committed:
Merge branch 'master' into issues/68932/remove-recursive

rschu1ze committed:
Disallow query cache for queries with non-throw overflow mode

Commit:
Merge pull request #69269 from alexkats/external-data-checksum

Use another error code for external data in buffer

marco-vb committed:
Implemented checks for new settings.

marco-vb committed:
Merge branch 'master' into marco-vb/setting-stop-insert-on-full-disk

joelynch committed:
Merge branch 'master' into joelynch/disk-encrypted-missing-method

marco-vb committed:
Merge branch 'ClickHouse:master' into marco-vb/x509-san-support-wildcard

rschu1ze committed:
Some fixups

kssenii committed:
Fix

kssenii committed:
Merge remote-tracking branch 'origin/s3queue-refactor' into s3queue-refactor

qoega committed:
Merge pull request #69253 from ClickHouse/restore-replace-to-null-settings-fix

Restore external tables to Null supports no settings

davenger committed:
Merge pull request #69383 from ClickHouse/tmp_files_lookup

Refactor temporary file usage in MergeTask with QueryPlan

davenger committed:
Merge branch 'master' into backport_mergetask

KevinyhZou committed:
Fix upgrade check failed for join_to_sort settings

rschu1ze committed:
Fix docs

Felixoid committed:
Add `kill_ci_runner` to ci_utils, will allow restarts

Felixoid committed:
Kill the runner process if integration tests fail to pre-pull

marco-vb committed:
Add new settings to settings change history.

rschu1ze committed:
Merge pull request #69549 from rschu1ze/qc-with-truncated-results

Disallow query cache for queries with non-throw overflow mode

rschu1ze committed:
Fix typo

rschu1ze committed:
Move ENABLE_LIBPQXX in a central place

nickitat committed:
log mark_segment_size on initiator

rschu1ze committed:
Add postgres as a submodule

rschu1ze committed:
Switch Postgres to 2f7bae2f92, adjust build description, delete libqp

Based on the code state of July 2021 which Kseniia copied over here:
https://github.com/ClickHouse/libpq/pull/5 (found out the hard way)

nickitat committed:
revive separate protocol versioning for PRs

rschu1ze committed:
3% more aesthetic build description

rschu1ze committed:
Remove obsolete target_include_directories (/config does not exist)

Avogar committed:
Keep original order of conditions during move to prewhere

nickitat committed:
Merge pull request #69554 from KevinyhZou/fix_settings_join_to_sort

Fix upgrade check failure for join_to_sort settings

Avogar committed:
Add test

nickitat committed:
Merge branch 'master' into adaptive_parallel_replicas

yakov-olkhovskiy committed:
Merge pull request #68236 from marco-vb/marco-vb/x509-san-support-wildcard

Added support for wildcard usage in x509 SubjectAltName identification.

kirillgarbar committed:
Merge remote-tracking branch 'kirillgarbar/master' into remove-stale-moving-parts

kssenii committed:
Merge pull request #69513 from ClickHouse/s3queue-refactor

S3Queue: small refactoring

Avogar committed:
Update tests

antonio2368 committed:
Merge pull request #69332 from ClickHouse/issues/68932/remove-recursive

Support removeRecursive natively in keeper

Avogar committed:
Remove bad files

Avogar committed:
Merge pull request #69147 from Avogar/fix-s3-cluster-structure

Fix propogating structure argument in s3Cluster

davenger committed:
Merge pull request #69525 from ClickHouse/backport_mergetask

Backport MergeTask changes from private to minimize merge conflicts

al13n321 committed:
Merge remote-tracking branch 'origin/master' into tomic

canhld94 committed:
Merge branch 'master' into parallel_join_limit_threads

Fix CI

taiyang-li committed:
allow empty inputs for arrayZip or arrayZipUnaligned

taiyang-li committed:
update uts

al13n321 committed:
Merge pull request #64953 from ClickHouse/tomic

Disallow creating refreshable MV on Linux < 3.15

taiyang-li committed:
fix style

arthurpassos committed:
Merge branch 'master' into multi_auth_methods

Michicosun committed:
enable removeRecursive in CI

nickitat committed:
Merge pull request #69406 from canhld94/parallel_join_limit_threads

Not retaining thread in concurrent hash join threadpool

rschu1ze committed:
Bump to v14.13

marco-vb committed:
Merge branch 'master' into marco-vb/setting-stop-insert-on-full-disk

antaljanosbenjamin committed:
Merge pull request #69301 from aohoyd/master

Add xml char escaping to docker's entrypoint

yariks5s committed:
Merge pull request #69514 from aiven-sal/aiven-sal/showcolumns

Correctly handle tables' names with dots in SHOW COLUMNS and SHOW INDEX

taiyang-li committed:
fix 01045_array_zip

taiyang-li committed:
fix 03230_array_zip_unaligned

taiyang-li committed:
Merge branch 'ClickHouse:master' into orc_dict_encode

divanik committed:
Merge pull request #66782 from a-a-f/input_format_json_empty_as_default

Add support for the `input_format_json_empty_as_default` setting

rschu1ze committed:
Fix GWP-asan crash

Michicosun committed:
fix tests

antonio2368 committed:
Support more advanced SSL options for Keeper internal communication

rschu1ze committed:
A better fix for #67476

Throw an exception instead of silently ignoring two conflicting settings.

arthurpassos committed:
update recently introduced test

arthurpassos committed:
fix black

kirillgarbar committed:
Set PRESERVE_BLOBS if part is fetched from another replica

mstetsyuk committed:
empty

antonio2368 committed:
Merge branch 'master' into keeper-better-ssl-support

tavplubix committed:
Merge pull request #69539 from ClickHouse/evillique-patch-2

Fix attach of ReplicatedMergeTree tables in Replicated databases

marco-vb committed:
Corrected implementation for check of new settings and fix lint of settings change history.

Avogar committed:
Try to fix tests

mstetsyuk committed:
sleep for 10ms beforen retrying to acquire a lock in DatabaseReplicatedDDLWorker::enqueueQueryImpl

antonio2368 committed:
Fixes

davenger committed:
Make dedup logic O(n*log(n)) instead of O(n^2)

davenger committed:
Fix clang-18 build

Felixoid committed:
Improve debug action

Felixoid committed:
Use local debug action in every action

Felixoid committed:
Merge steps together to minimize grouping

nickitat committed:
impl

nickitat committed:
Update tests/integration/test_parallel_replicas_snapshot_from_initiator/test.py

Co-authored-by: Igor Nikonov <954088+devcrafter@users.noreply.github.com>

nickitat committed:
fix

mstetsyuk committed:
use base/sleep.h

nickitat committed:
Merge pull request #69404 from bigo-sg/69135

`have_compressed` is lost in `HashJoin::reuseJoinedData`

mstetsyuk committed:
Merge pull request #69274 from ClickHouse/fix-metadata-version-in-zookeeper

fix `metadata_version` in ZooKeeper

marco-vb committed:
Added integration testing for newly implemented settings.

marco-vb committed:
Merge branch 'ClickHouse:master' into marco-vb/setting-stop-insert-on-full-disk

davenger committed:
Check time limit while analyzing indexes

marco-vb committed:
Black formatting python test.

nickitat committed:
fix

vitlibar committed:
Masking sensitive info in gcs() table function.

yakov-olkhovskiy committed:
unification of FunctionSecretArgumentsFinder

yakov-olkhovskiy committed:
fix

yakov-olkhovskiy committed:
fix tidy

nickitat committed:
Merge pull request #69602 from ClickHouse/rm_explicit_initial_announce_from_RFMT

Remove explicit announce from local replica in ReadFromMergeTree

nickitat committed:
Merge branch 'master' into adaptive_parallel_replicas

nickitat committed:
fix build

marco-vb committed:
Only make checks to stop inserts if settings are being used.

marco-vb committed:
Reduced disk size on test for faster execution.

taiyang-li committed:
Merge branch 'master' into orc_dict_encode

rschu1ze committed:
Merge pull request #69583 from rschu1ze/better-fix-67476

A better fix for #67476

arthurpassos committed:
trigger ic

rschu1ze committed:
Fix sizing with unconstrained thread pool size

antonio2368 committed:
Merge branch 'master' into keeper-better-ssl-support

antonio2368 committed:
Fix

vitlibar committed:
Merge pull request #69611 from vitlibar/masking-sensitive-info-in-gcs-table-function

Masking sensitive info in gcs() table function

vdimir committed:
Merge pull request #69075 from kirillgarbar/remove-stale-moving-parts

Remove stale moving parts without zookeeper

Olexandr88 committed:
Docs: Update index

rschu1ze committed:
Merge pull request #69493 from ucasfl/vector-index-insert

Speedup insert performance of vector similarity index by parallelization

antonio2368 committed:
Merge pull request #69578 from ClickHouse/issues/68932/enable_in_ci

enable removeRecursive in CI

vdimir committed:
fix

Olexandr88 committed:
Docs: Update index

yariks5s committed:
empty commit (I've changed the changelog entry)

antonio2368 committed:
Fix Keeper multi request preprocessing with NOAUTH

Felixoid committed:
Merge pull request #69599 from ClickHouse/local-debug

Improve debug step in actions

antaljanosbenjamin committed:
Update filelog.md

yariks5s committed:
init

davenger committed:
Failpoint for testing slow index analysis

davenger committed:
Test for checking time limit in index analysis

davenger committed:
Merge branch 'master' into check_time_limit_in_index_analysis

yariks5s committed:
fix references + remove index

Avogar committed:
Update docs of JSONAsObject format with new JSON type

yakov-olkhovskiy committed:
Merge branch 'master' into refactor-secret-finder

vdimir committed:
Merge pull request #68595 from ClickHouse/vdimir/fix_function_printf_style

Fix style in Functions/printf.cpp

yakov-olkhovskiy committed:
fix after master merge

thevar1able committed:
Fix

rfraposa committed:
Merge pull request #69622 from Olexandr88/patch-1

Docs: Update index

yakov-olkhovskiy committed:
fix

thevar1able committed:
Merge pull request #69642 from ClickHouse/fix-invalid-clickhouse-format-invocation

Fix invalid `clickhouse-format` invocation in tests

yariks5s committed:
Merge pull request #69625 from ClickHouse/fix-filelog-docs

Update filelog.md

Avogar committed:
Merge pull request #69635 from ClickHouse/Avogar-patch-7

Update docs of JSONAsObject format with new JSON type

rschu1ze committed:
Merge pull request #69581 from rschu1ze/remove-n

Remove superfluous `--multiquery/-n`, pt. V

mstetsyuk committed:
Merge pull request #69588 from ClickHouse/add-sleep-for-in-enqueueQueryImpl

sleep for 50ms before retrying to acquire a lock in `DatabaseReplicatedDDLWorker::enqueueQueryImpl`

thevar1able committed:
Fix

thevar1able committed:
temp-commit

pufit committed:
Merge pull request #65277 from arthurpassos/multi_auth_methods

Multi auth methods

thevar1able committed:
Merge pull request #69601 from ClickHouse/fix-tsan-writebufferfromhttpserverresponse

Try to fix data race in `WriteBufferFromHTTPServerResponse`

thevar1able committed:
Merge pull request #69655 from ClickHouse/try-fix-02447_drop_database_replica

Try fix `02447_drop_database_replica`

antonio2368 committed:
Merge pull request #69627 from ClickHouse/keeper-fix-multi-no-auth

Fix Keeper multi request preprocessing with NOAUTH

KochetovNicolai committed:
Merge pull request #69298 from ClickHouse/array-join-step-refactoring

Refactor ArrayJoin step.

antonio2368 committed:
Merge branch 'master' into keeper-better-ssl-support

yariks5s committed:
Merge pull request #69626 from yariks5s/small_tests_fix

Reorder some tests (follow-up to #69514)

rschu1ze committed:
Merge pull request #69564 from rschu1ze/untrash-libpq

Replace libpq code dump by postgresql fork and bump to v14.3

mstetsyuk committed:
sync changes to InterpreterDropQuery::executeToDatabaseImpl from the private fork

vdimir committed:
Merge pull request #69328 from ClickHouse/vdimir/integration-tests-randomize-settings

Randomize integration tests settings

vitlibar committed:
Corrections after review.

kssenii committed:
Merge pull request #68592 from Sergey2Gnezdilov/patch-1

Update nats.md

kssenii committed:
Merge pull request #68696 from ucasfl/minor

Use proper ErrorCodes, replace NETWORK_ERROR by HDFS_ERROR

vitlibar committed:
Fix test.

kssenii committed:
Merge pull request #68821 from joelynch/joelynch/disk-encrypted-missing-method

Fix zero copy bug with encrypted disk and UNFREEZE

kssenii committed:
Merge pull request #69092 from 1on/master

Ability to limit columns for tables in MaterializedPostgreSQL

yakov-olkhovskiy committed:
Merge branch 'master' into refactor-secret-finder

kssenii committed:
Update assert

rschu1ze committed:
Bump to v18.8

robot-clickhouse committed:
Automatic style fix

NikBarykin committed:
Fix CE

davenger committed:
Merge pull request #69606 from ClickHouse/check_time_limit_in_index_analysis

Check time limits while analyzing indexes

Avogar committed:
Merge pull request #68591 from bigo-sg/orc_dict_encode

Add settings `output_format_orc_dictionary_key_size_threshold` to allow user to enable dict encoding for string column in ORC output format

kssenii committed:
Merge pull request #68520 from ClickHouse/fix-bad-exception-messages

Fix bad exception messages

antonio2368 committed:
Merge pull request #69582 from ClickHouse/keeper-better-ssl-support

Support more advanced SSL options for Keeper internal communication

nickitat committed:
Merge pull request #68424 from ClickHouse/adaptive_parallel_replicas

Adaptive mark_segment_size for parallel replicas

davenger committed:
Verify that there are no intersecting parts in the resulting all_parts_to_read

davenger committed:
More optimal check for intrsecting parts in DefaultCoordinator init

vdimir committed:
Merge pull request #67966 from ClickHouse/vdimir/datetime64_constant_to_ast_f

Add test cases to 03217_datetime64_constant_to_ast

davenger committed:
Fix for using part after std::move from it

davenger committed:
Use adjacent_find to check adjacent parts

rschu1ze committed:
Prohibit ALTER TABLE ... ADD INDEX ... TYPE inverted if setting = 0

mstetsyuk committed:
Merge pull request #69670 from ClickHouse/sync-executeToDatabaseImpl-with-private-fork

sync changes to `InterpreterDropQuery::executeToDatabaseImpl` from the private fork

Felixoid committed:
Merge pull request #69557 from ClickHouse/integration-prepull-kill-runner

Kill runner when integration tests fail to pre-pull

vitlibar committed:
Merge pull request #69346 from vitlibar/restore-access-dependencies

Improve restoring of access entities' dependencies

Avogar committed:
Merge pull request #69560 from Avogar/fix-prewhere-reorder

Keep original order of conditions during move to prewhere

vitlibar committed:
Merge pull request #69201 from NikBarykin/allow_arguments_in_custom_database_engines

Allow custom settings in database engine

jsc0218 committed:
Merge pull request #69376 from marco-vb/marco-vb/setting-stop-insert-on-full-disk

Add user-level settings min_free_diskspace_bytes_to_throw_insert and min_free_diskspace_ratio_to_throw_insert

Algunenano committed:
Revert "Add user-level settings min_free_diskspace_bytes_to_throw_insert and min_free_diskspace_ratio_to_throw_insert"

Algunenano committed:
Merge pull request #69705 from ClickHouse/revert-69376-marco-vb/setting-stop-insert-on-full-disk

Revert "Add user-level settings min_free_diskspace_bytes_to_throw_insert and min_free_diskspace_ratio_to_throw_insert"

kssenii committed:
Merge pull request #69673 from ClickHouse/update-assert

Update assert

davenger committed:
Merge pull request #69596 from ClickHouse/fix_dedup_in_parallel_replicas_announcement

Optimize complexity of part deduplication in parallel replicas announcement

rschu1ze committed:
Merge pull request #69684 from rschu1ze/disallow-alter-table-add-inv-idx

Prohibit `ALTER TABLE ... ADD INDEX ... TYPE` inverted if setting = 0

Commit:
Merge pull request #69576 from bigo-sg/arrayzip_allow_empty

Allow empty arguments for arrayZip/arrayZipUnaligned

tbragin committed:
Update README.md - Meetups

yakov-olkhovskiy committed:
Merge pull request #69615 from ClickHouse/refactor-secret-finder

Unification of FunctionSecretArgumentsFinder

rschu1ze committed:
Merge pull request #69674 from rschu1ze/bump-pg

Bump libpq from v14.3 to v15.8

kssenii committed:
Merge pull request #69714 from tbragin/patch-15

Update README.md - Meetups

max-vostrikov committed:
Merge remote-tracking branch 'origin/test_postgres_kill' into test_postgres_kill

max-vostrikov commented:
a bit to much difference to merge. I'll crate a new PR for this
